### PR TITLE
Consistently use ScriptInfo for converting positions to Locations

### LINKED ISSUE #24857: "rename file" and "move to new file" refactors do not work when using `files`
**TypeScript Version:**  2.9.1

**Search Terms:** 
files "move to new file" "rename file"

**Code**

```json
// tsconfig.json
{
  "files": ["./src/index.ts"]
}
```

``` ts
// ./src/index.ts
import foo from './other';
```

``` ts
// ./src/other.ts
export default {}

const bar = {};
```

**Expected behavior:**
- When renaming the file `./src/other.ts`, I expect TypeScript/VSCode to prompt me to update the import in `./src/index.ts`.
- When selecting `bar` in `./src/other.ts`, TypeScript/VSCode offers to move this to a new file. When actioning this, I expect this to work.

(TypeScript knows that `./src/other.ts` is in the project because it is part of the traced dependency graph from the entry file `./src/index.ts`.)

**Actual behavior:**
- When renaming the file `./src/other.ts`, there is no prompt to update the import.
- When selecting `bar` in `./src/other.ts`, TypeScript/VSCode offers to move this to a new file. When actioning this, nothing happens.

**Playground Link:**  

**Related Issues:** 
https://github.com/Microsoft/vscode/issues/51366
https://github.com/Microsoft/TypeScript/pull/23573
https://github.com/Microsoft/TypeScript/pull/23726

---

A known workaround is to use the `include` option instead of `files`. However, I am curious why this is necessary. I prefer to list the entry file and let TypeScript work the rest out. This is especially helpful when a folder has a combination of source and test files—TypeScript automatically knows not to include the test files because they are not part of the dependency graph.

If there is a good reason why these commands/refactors can't work with `files`, could we make the experience clearer? Why is the "move to new file" command suggested in VSCode when it leads to nothing? If so, perhaps this is an issue for VSCode.

### LINKED ISSUE #25560: Quick fix triggers debug assert
Create an anonymous buffer

```ts
let foo = 1;
fooo/**/
```

Get quick fixes at `fooo`:

```
[Error  - 12:55:44 PM] 'getCodeFixes' request failed with error.
Error processing request. Debug Failure. False expression.
Error: Debug Failure. False expression.
    at mapTextChangesToCodeEdits (C:\Users\Daniel\AppData\Local\nvs\node\10.4.1\x64\node_modules\typescript\lib\tsserver.js:119251:22)
    at IOSession.Session.mapTextChangeToCodeEdit (C:\Users\Daniel\AppData\Local\nvs\node\10.4.1\x64\node_modules\typescript\lib\tsserver.js:119077:24)
    at C:\Users\Daniel\AppData\Local\nvs\node\10.4.1\x64\node_modules\typescript\lib\tsserver.js:119074:73
    at Array.map (native)
    at IOSession.Session.mapTextChangesToCodeEdits (C:\Users\Daniel\AppData\Local\nvs\node\10.4.1\x64\node_modules\typescript\lib\tsserver.js:119074:36)
    at IOSession.Session.mapCodeFixAction (C:\Users\Daniel\AppData\Local\nvs\node\10.4.1\x64\node_modules\typescript\lib\tsserver.js:119070:84)
    at C:\Users\Daniel\AppData\Local\nvs\node\10.4.1\x64\node_modules\typescript\lib\tsserver.js:119023:96
    at Array.map (native)
    at IOSession.Session.getCodeFixes (C:\Users\Daniel\AppData\Local\nvs\node\10.4.1\x64\node_modules\typescript\lib\tsserver.js:119023:55)
    at Session.handlers.ts.createMapFromTemplate._a.(anonymous function) (C:\Users\Daniel\AppData\Local\nvs\node\10.4.1\x64\node_modules\typescript\lib\tsserver.js:117896:61)
    at C:\Users\Daniel\AppData\Local\nvs\node\10.4.1\x64\node_modules\typescript\lib\tsserver.js:119186:88
    at IOSession.Session.executeWithRequestId (C:\Users\Daniel\AppData\Local\nvs\node\10.4.1\x64\node_modules\typescript\lib\tsserver.js:119177:28)
    at IOSession.Session.executeCommand (C:\Users\Daniel\AppData\Local\nvs\node\10.4.1\x64\node_modules\typescript\lib\tsserver.js:119186:33)
    at IOSession.Session.onMessage (C:\Users\Daniel\AppData\Local\nvs\node\10.4.1\x64\node_modules\typescript\lib\tsserver.js:119206:35)
    at Interface.<anonymous> (C:\Users\Daniel\AppData\Local\nvs\node\10.4.1\x64\node_modules\typescript\lib\tsserver.js:120473:27)
    at emitOne (events.js:96:13)
    at Interface.emit (events.js:191:7)
    at Interface._onLine (readline.js:241:10)
    at Interface._normalWrite (readline.js:384:12)
    at Socket.ondata (readline.js:101:10)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:191:7)
    at readableAddChunk (_stream_readable.js:178:18)
    at Socket.Readable.push (_stream_readable.js:136:10)
    at Pipe.onread (net.js:560:20)
```

### PR DESCRIPTION
**Pull request by andy-ms**
@DanielRosenwasser Could you check if this fixes #25560 ?
Should fix #24857 too

### CODE REVIEW

**Comment by andy-ms** [comment]
Waiting for review from @DanielRosenwasser

File: src/server/session.ts [hunk 1]
```diff
@@ -1824,24 +1824,20 @@ namespace ts.server {
...
         }
 
-        private mapTextChangeToCodeEdit(project: Project, change: FileTextChanges): protocol.FileCodeEdits {
-            return mapTextChangesToCodeEditsForFile(change, project.getSourceFileOrConfigFile(this.normalizePath(change.fileName)));
-        }
-
-        private mapTextChangeToCodeEditUsingScriptInfoOrConfigFile(change: FileTextChanges): protocol.FileCodeEdits {
-            return mapTextChangesToCodeEditsUsingScriptInfoOrConfig(change, this.projectService.getScriptInfoOrConfig(this.normalizePath(change.fileName)));
+        private mapTextChangeToCodeEdit(change: FileTextChanges): protocol.FileCodeEdits {
+            return mapTextChangesToCodeEdits(change, this.projectService.getScriptInfoOrConfig(this.normalizePath(change.fileName)));
```
Comment (sheetalkamat) [inline src/server/session.ts]: I dont think you want to convert the fileName to normalizePath.. Instead you want to pass it as is to the getScriptOfConfig and there you will do the right thing.

Comment (sheetalkamat) [inline src/server/session.ts]: Add test for the scenario @DanielRosenwasser  was seeing where in the buffer was buffer with dynamic file name

**Comment by mhegazy** [comment]
@andy-ms can you please port this to release-3.0 as well.

**Pull request merged**